src/main/java/SpringJPA/PageController.java

```diff
@@ -99,6 +109,41 @@ public String makeAPICall(Principal principal, @ModelAttribute String placeholde
         userRepository.save(user);
         return "redirect:/user";
     }
+
+    @PostMapping("/user/FreeTrialRequest")
+    public String makeFreeTrialRequest(Principal principal, @ModelAttribute String placeholder){
+        User user = userRepository.findByUsername(principal.getName());
+        if (user.getRole() == UserType.ROLE_NONTRIAL && !user.getHasUsedTrial()){
+            user.setRole(UserType.ROLE_TRIAL);
+            user.startTrial();
+            SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRole().toString());
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            List<GrantedAuthority> updatedAuthorities = new ArrayList<>(auth.getAuthorities());
+            updatedAuthorities.add(authority);
+            Authentication newAuth = new UsernamePasswordAuthenticationToken(auth.getPrincipal(), auth.getCredentials(), updatedAuthorities);
+            SecurityContextHolder.getContext().setAuthentication(newAuth);
+        }
+        userRepository.save(user);
+        return "redirect:/user";
+    }
+
+    @PostMapping("/user/PaidUserRequest")
+    public String makePaidUserRequest(Principal principal, @ModelAttribute String placeholder){
+        User user = userRepository.findByUsername(principal.getName());
+        if (user.getRole() == UserType.ROLE_NONTRIAL || user.getRole() == UserType.ROLE_TRIAL){
+            user.setRole(UserType.ROLE_PREMIUM);
+            SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRole().toString());
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            List<GrantedAuthority> updatedAuthorities = new ArrayList<>(auth.getAuthorities());
+            updatedAuthorities.add(authority);
+            Authentication newAuth = new UsernamePasswordAuthenticationToken(auth.getPrincipal(), auth.getCredentials(), updatedAuthorities);
+            SecurityContextHolder.getContext().setAuthentication(newAuth);
```

Comment:
I noticed some code repetition here, perhaps it should be placed into a function.

Reply:
pushed an update